app/controllers/search_controller.rb

```diff
@@ -10,15 +10,15 @@ def show
     service = Service.new(params[:endpoint])
 
     result = service.query(params_q_scrubbed)
-    @presenter = SearchPresenter.new(service, result)
+    @presenter = SearchPresenter.new(service, result, params_q_scrubbed)
```

Comment:
In this old approach, the search executes before the presenter is initialized.  I rather prefer this way. The presenter doesn't have to know anything about how to do the search, only how to display the results.